### PR TITLE
Controller add module method

### PR DESCRIPTION
**Pull request by bannerintheUK**

### CODE REVIEW

File: packages/cerebral/src/Controller.js [hunk 1]
```diff
@@ -196,6 +196,77 @@ class Controller extends FunctionTree {
...
+
+  addModule (path, module) {
+    const pathArray = path.split('.')
+    const moduleKey = pathArray.pop()
+
+    const parentModule = pathArray.reduce((currentModule, key) => {
+      if (!currentModule) {
+        throwError(`The path "${pathArray.join('.')}" is invalid, can not add module. Does the path "${pathArray.splice(0, path.length - 1).join('.')}" exist?`)
+      }
+      console.log(currentModule.modules[key])
```
Comment (christianalfoni) [inline packages/cerebral/src/Controller.js]: Ops, this should not be here? :)

File: packages/cerebral/src/Controller.js [hunk 2]
```diff
@@ -196,6 +196,77 @@ class Controller extends FunctionTree {
...
+    this.model.unset(pathArray)
+
+    if (module.provider) {
+      this.contextProviders.splice(this.contextProviders.indexOf(module.provider), 1)
+    }
+
+    this.flush()
+  }
+
+  getModule (path) {
```
Comment (christianalfoni) [inline packages/cerebral/src/Controller.js]: Remove this method. Do not want people to start grabbing modules :)

File: packages/cerebral/src/Controller.js [hunk 3]
```diff
@@ -196,6 +196,77 @@ class Controller extends FunctionTree {
...
+    const moduleKey = pathArray.pop()
+
+    const parentModule = pathArray.reduce((currentModule, key) => {
+      return currentModule.modules[key]
+    }, this)
+
+    return parentModule.module.modules[moduleKey]
+  }
+
+  getModules () {
```
Comment (christianalfoni) [inline packages/cerebral/src/Controller.js]: Remove this method as well :)

File: packages/cerebral/src/Controller.test.js [hunk 4]
```diff
@@ -254,4 +254,63 @@ describe('Controller', () => {
     })
     controller.getSignal('test')()
   })
+  it('should expose method to getModules', () => {
```
Comment (christianalfoni) [inline packages/cerebral/src/Controller.test.js]: Do not need test for this as method should be removed

File: packages/cerebral/src/Controller.test.js [hunk 5]
```diff
@@ -254,4 +254,63 @@ describe('Controller', () => {
...
+    })
+    assert.deepEqual(controller.getModules(), {
+      foo: {
+        modules: {},
+        signals: {},
+        state: {}
+      }
+    })
+  })
+  it('should expose method to getModule using path', () => {
```
Comment (christianalfoni) [inline packages/cerebral/src/Controller.test.js]: Same here

**Comment by Guria** [comment]
@christianalfoni is it all ok now?

**Comment by Guria** [comment]
@bannerintheUK can you add docs to api section as well?

**Comment by christianalfoni** [comment]
@henri-hulski Updated :)

**Pull request merged**